src/routing/staticSite.js

```diff
@@ -1,27 +1,6 @@
 import * as endpoints from '../endpoints/index.js';
 
-
-/** Gatsby static HTML pages and other assets.
- *
- * Any URLs matching Gatsby's static files will be handled here, e.g.
- * static-site/public/influenza/index.html will be served for /influenza.
- *
- * When a Gatsby dev server is in use, assets are proxied from the dev server
- * instead of served straight from disk.
- */
 export async function setup(app) {
-  if (app.locals.gatsbyDevUrl) {
-    const {createProxyMiddleware} = await import("http-proxy-middleware");
-
-    // WebSocket endpoint
-    app.get("/socket.io/", createProxyMiddleware(app.locals.gatsbyDevUrl, { ws: true }));
-
-    /* This is the end of the line in gatsbyDevUrl mode, as the proxy middleware
-     * doesn't fallthrough on 404 and even if it did the Gatsby dev server
-     * returns a 200 Ok for missing pages.
-     */
-    app.use(createProxyMiddleware(app.locals.gatsbyDevUrl));
-  } else {
-    app.use(endpoints.static.gatsbyAssets);
-  }
-}
+  app.routeAsync("*")
+    .getAsync(endpoints.nextJsApp.handleRequest);
+}
```

Comment:
In trying to understand this setup better, I searched for examples of Express servers using Next.js. I didn't find any, but found [this example](https://github.com/vercel/next.js/blob/e969d226999bb0fcb52ecc203b359f3715ff69bf/examples/custom-server-express/server.js) in the official repo of a Next.js server using Express (though there are some [unanswered questions](https://github.com/vercel/next.js/pull/24814#pullrequestreview-652292574)).

Reply:
One way to think about it is we have two largely independent layers:

* Layer 1 is our express server with defined routes and handlers.
* Layer 2 is our nextjs server which handles any route route not handled by layer 1.

In the previous implementation is was largely similar except Layer 2 was an [express static-file server](https://expressjs.com/en/starter/static-files.html) which would access the pre-built Gatsby assets. There are a few other differences for sure, including 404 handling¹. Let me know if you have specific questions!

¹ _One of the strangest parts of the previous set up was when the server would explicitly send a 404 page which would then use client-side routing and show the user another page altogether!_

Reply:
Thanks for the overview, that makes sense. I'm just wondering if there is any reason to switch the entrypoint from Express to Next.js (example linked above is set up like that).

Reply:
I tried that initially but switched to the current implementation for a few reasons. Firstly, for development mode using a nextjs server is very compelling (hot reloading etc), and using a similar approach in production seemed less prone to bugs. Secondly, for routes such as `/ncov/*` (where `ifDatasetExists` is false, so we aren't using Auspice), I wanted to move us away from the previous approach of sending a 404 page and relying on client-side routing to route us to the right page (there are other ways to solve this, but they felt brittle). Finally, using a next.js server opens the door to different forms of SSR should we want them, although we should be able to do true SSG without it.

Happy to explore this more if you find that it's clearer how the routing works under that model?

> example linked above is set up like that

The example is essentially the same as our current Layer 2. (Wasn't clear if you were claiming that the example was similar to our current approach or was like our previous approach.)

Reply:
> The example is essentially the same as our current Layer 2

Forgot to update here - had a quick call with @jameshadfield where he gave a good explanation. The combination of `const prepared = nextJsApp.prepare()` + `await prepared` in [src/endpoints/nextjs.js](https://github.com/nextstrain/nextstrain.org/blob/1b0df7748055bf8525d503e70c33a5204ef1c3e2/src/endpoints/nextjs.js) is functionally equivalent to `app.prepare().then(…` in the Next.js example.